### DISCUSSION TITLE
Modeling a recommendation stack call with Hamilton

### DISCUSSION THREAD

**Discussion by skrawcz** [stitchfix/hamilton#206] [Ideas]
Idea: Could you use Hamilton to model a recommendation stack call

Steps:
1. Define a Hamilton DAG
2. Compile it to services [optional] (via a driver?)
3. Define a driver that knows functions to services.
4. Execute the driver and it will resolve walking the DAG across a bunch of services.

Discuss your thoughts here -- what should the requirements be?

**Comment by jakemannix** [comment]
Simple *synchronous* model inference DAGs would be a good testing ground: can Hamilton help a user run a DAG with just a couple of critical typical ML inference nodes: [preprocess HTTP/gRPC request] -> [fetch data from remote Feature Store endpoint] -> [send model inference request to remote model hosting service endpoint] -> [post-process model outputs and produce appropriate HTTP/gRPC response]

Obviously more complex DAGs need to work, but this "hello world" would showcase how synchronous requests could be wired into Hamilton
